src/logging_vnode.erl

```diff
@@ -75,9 +77,16 @@ asyn_read_from(Preflist, Log, From) ->
 %% @doc synchronous read_from operation
 -spec read_from({partition(), node()}, log_id(), op_id()) -> term().
 read_from(Node, LogId, From) ->
-    riak_core_vnode_master:sync_command(Node,
+    case riak_core_vnode_master:sync_command(Node,
                                         {read_from, LogId, From},
-                                        ?LOGGING_MASTER).
+                                        ?LOGGING_MASTER) of
+        {ok, []} ->
+            {ok, []};
+        {ok, [H|T]} ->
+            {ok, [H|T]};
+        {error, Reason} ->
```

Comment:
This can be simplified, there is no need to pattern match here, or is there one?